services/ui-src/src/utils/other/mask.tsx

```diff
@@ -1,104 +1,88 @@
+
+  // If entire value is not 0, remove all leading zeros
+  if (parseFloat(maskedValue) !== 0) {
     // Remove all leading zeroes if value is not equal to 0
-    value = value.replace(/^0+/g, "");
+    maskedValue = maskedValue.replace(/^0+/g, "");
   }
+
   // Convert String to a float to begin operation
-  const valueAsFloat = parseFloat(value);
+  const valueAsFloat = parseFloat(maskedValue);
+
   // Slide any extra decimals down to 2
   const fixedDecimal = valueAsFloat.toFixed(2);
-  // Clean up the float value and add in commas to delineate thousands if needed
-  const cleanedValue = Number(fixedDecimal).toLocaleString("en");
-  return cleanedValue.toString();
+
+  // Add in commas to delineate thousands (if needed)
+  maskedValue = Number(fixedDecimal).toLocaleString("en");
+  return { maskedValue: maskedValue.toString(), cleanedValue: cleanedValue };
 }
 
 /**
  * Splits a string with a : and converts each side to a comma-separated string
  * @param {String} value
  * @returns {String}
  */
-export function convertToCommaSeparatedRatioString(value: string): string {
-  // Remove all characters except digits and decimal points.
-  value = value.replace(/[^\d.:]/g, "");
+export function convertToThousandsSeparatedRatioString(value: string): {
+  maskedValue: string;
+  cleanedValue: string;
+} {
+  // Clean value
+  const cleanedInput = cleanRatioInput(value);
+  if (!cleanedInput.isValid) return { maskedValue: value, cleanedValue: value };
 
   // Grab the left and right side of the ratio sign
-  const values = value.split(":");
-
-  // Begin creating the final output
-  let cleanedValue = "";
+  const values = cleanedInput.cleanedValue.split(":");
 
   // Create the left side of the output and make the number (if provided) pretty
-  if (values[0] != "") cleanedValue += convertToCommaSeparatedString(values[0]);
-  else cleanedValue += "";
-
-  // Put in the ratio sign in the middle of the two numbers
-  cleanedValue += ":";
+  values[0] = convertToThousandsSeparatedString(values[0]).maskedValue;
 
   // Create the right side of the output and make the number (if provided) pretty
-  if (values.length >= 2 && values[1] != "")
-    cleanedValue += convertToCommaSeparatedString(values[1]);
-  else cleanedValue += "";
+  values[1] = convertToThousandsSeparatedString(values[1]).maskedValue;
 
-  return cleanedValue;
+  const maskedValue = values.join(":");
+  return { maskedValue: maskedValue, cleanedValue: cleanedInput.cleanedValue };
 }
 
-/**
- * Converts string to the appropriate custom masked format
- * @param {String} value
- * @returns {String}
- */
-export function maskValue(
+// if valid custom mask, return masked value; else return value
+export const applyMask = (
   value: string,
-  mask: keyof typeof customMaskMap
-): string {
-  const selectedCustomMask = customMaskMap[mask];
-  if (isValidNumericalString(value)) {
-    return selectedCustomMask(value);
-  } else return value;
-}
+  maskName?: keyof typeof maskMap | null
```

Comment:
What happens if `maskName` is undefined here?

Reply:
it defaults to the thousand separated mask!